Add Spark plugin

amousavigourabi opened the pull request:
Add a Spark plugin to generate WTA traces directly from Spark runs.

amousavigourabi committed:
Adds more stream operations

amousavigourabi committed:
Add reduce, count, and collection constructor

qutianchen committed:
doc review

Commit:
Merge branch '148-pyspark-support-for-the-plugin' into 'dev'

Resolve "PySpark support for the plugin"

Closes #148

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!96

qutianchen committed:
docs

amousavigourabi committed:
Move static guard in ParquetSchema

qutianchen committed:
checkstyle

amousavigourabi committed:
Fix static guard

amousavigourabi committed:
Update CHANGELOG

amousavigourabi committed:
Merge remote-tracking branch 'origin/176-streaming' into 176-streaming

amousavigourabi committed:
Merge branch '182-fix-resource-state-parquet-schema-gen-err' into 'dev'

Resolve "Fix resource state parquet schema gen err"

Closes #182

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!102

amousavigourabi committed:
Removes PITest

amousavigourabi committed:
Merge branch '186-remove-mutation-testing-from-pipeline' into 'dev'

Resolve "Remove mutation testing from pipeline"

Closes #186

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!103

qutianchen committed:
Merge remote-tracking branch 'origin/dev' into 175-critical-path-retrieval-dag-sorting-for-finding-longest-path

amousavigourabi committed:
Merge branch 'dev' into 176-streaming

qutianchen committed:
docs on additional nodes

amousavigourabi committed:
Expands the Stream API

Makes streams cloneable and adds drop

amousavigourabi committed:
Fix Stream#reduce bug

Would throw NoSuchElementExceptions when we did not want it to, now its behaviour is as the java Stream one

amousavigourabi committed:
Adds streams for tasks

Resource streams are not yet implemented

amousavigourabi committed:
Add convenience methods for Stream and KeyedStream

amousavigourabi committed:
Fix Stream#clone

amousavigourabi committed:
Adds Streams for resources as well

amousavigourabi committed:
Update CHANGELOG

Commit:
add log for command

Commit:
fix mr conflicts

amousavigourabi committed:
Cleans up some code

Commit:
make exceptions more specific add checks to see if resources are available and modify tests

amousavigourabi committed:
Fixes reduce

Commit:
fix tests

Commit:
fix perf tests

Commit:
update readme

Commit:
fix changelog

Commit:
combine logs

Commit:
review fixes

Commit:
fix sentence

lyadalachanchu20033 committed:
Merge branch '181-remove-error-logging-for-shell-commands' into 'dev'

Resolve "Remove error logging for shell commands"

Closes #181

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!100

amousavigourabi committed:
Fixes clone

amousavigourabi committed:
Make clone overly complicated

checkstyle does not approve, it works though

amousavigourabi committed:
Cleans up Stream#clone

amousavigourabi committed:
Refactors the cloning code

amousavigourabi committed:
Merge remote-tracking branch 'origin/dev' into 176-streaming

amousavigourabi committed:
Implement Cloneable

amousavigourabi committed:
Resolve on disk storage issues

amousavigourabi committed:
Changes some primitives to wrappers

amousavigourabi committed:
Makes Stream nodes serialize

This allows us to serialize streams inside of other streams, which we need

amousavigourabi committed:
Correctly output ResourceStates

qutianchen committed:
map name

amousavigourabi committed:
Optimizes the serialization

Now instead of serializing 1800! objects, we only serialize 1800, orders upon orders of magnitude of improved performance

qutianchen committed:
naming conventions

qutianchen committed:
Merge remote-tracking branch 'origin/dev' into 175-critical-path-retrieval-dag-sorting-for-finding-longest-path

amousavigourabi committed:
Removes Streamed Streams

amousavigourabi committed:
Apply Spotless

amousavigourabi committed:
Revert "Optimizes the serialization"

qutianchen committed:
Merge branch '175-critical-path-retrieval-dag-sorting-for-finding-longest-path' into 'dev'

Resolve "critical path retrieval: DAG sorting for finding longest path"

Closes #175

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!94

amousavigourabi committed:
Resolve serialization failures

amousavigourabi committed:
Clean up the code

amousavigourabi committed:
Resolves Stream serializing edge cases

amousavigourabi committed:
Fix the edge case fix to resolve heap err

amousavigourabi committed:
Add asynchronous operations for serialization

1henrypage committed:
logs

amousavigourabi committed:
Fix thread run

Commit:
convert basetrace fields to primitives

Commit:
adjust tests

Commit:
Merge branch '176-streaming' of gitlab.ewi.tudelft.nl:cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev into 176-streaming

amousavigourabi committed:
Fix users and groups

1henrypage committed:
make the logs in E2E

Commit:
stub isAvailable

Commit:
add setter

1henrypage committed:
finished

1henrypage committed:
log stage or task level metrics

1henrypage committed:
change log level instructions

lyadalachanchu20033 committed:
Merge branch '188-increase-proc-test-coverage' into 'dev'

Resolve "Increase Proc Test Coverage"

Closes #188

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!106

1henrypage committed:
Merge branch '149-add-logging-statements-to-the-source-code' into 'dev'

Resolve "Add logging statements to the source code"

Closes #149

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!88

amousavigourabi committed:
Cleans up the codebase

amousavigourabi committed:
Resolves workflow bug

amousavigourabi committed:
Adds javadoc

amousavigourabi committed:
Write from Stream instead of List

amousavigourabi committed:
Moves away from the shutdown callback

amousavigourabi committed:
Add Javadoc

amousavigourabi committed:
Optimizes use of the ExecutorService

amousavigourabi committed:
Fix Parquet schema generation bug

amousavigourabi committed:
Fix ParquetSchema bug pt.2

amousavigourabi committed:
Resolves some test failures

amousavigourabi committed:
Updates application listener tests

amousavigourabi committed:
Switch to ThreadPoolExecutor

amousavigourabi committed:
Linters

amousavigourabi committed:
Adds thread cleanup to tests

amousavigourabi committed:
Cleans up the code

amousavigourabi committed:
Optimizes Stream#copy

amousavigourabi committed:
Optimizes job level listener

amousavigourabi committed:
Moves the moment we create folders

For usability, users might delete the output directory after starting the run, which would cause the writes to fail

amousavigourabi committed:
Merge branch 'dev' into 176-streaming

amousavigourabi committed:
Resolves conflicts

amousavigourabi committed:
Minor clean up

amousavigourabi committed:
Fixes workflows not showing tasks

amousavigourabi committed:
Make code using KeyedStream less bug-prone

amousavigourabi committed:
Abstracts away some code

Moves code from ApplicationLevelListener to the SparkDataSource

Commit:
fix job level listener test

amousavigourabi committed:
Adds await at shutdown

Commit:
fix task level listener test

Commit:
attempt test fixes

amousavigourabi committed:
Simplifies tests

amousavigourabi committed:
Removes listener removal from Plugin

It's all located in the listeners now

Commit:
fix listener integration tests

Commit:
attempt task level integration test fix

amousavigourabi committed:
Unit tests Stream convenience methods

Commit:
fix tests

amousavigourabi committed:
Help the loss of task-workflow assocs

Commit:
remove thread pool await

amousavigourabi committed:
Creates separate executor services instead of blocking

amousavigourabi committed:
Revert "Creates separate executor services instead of blocking"

Commit:
fix tests

amousavigourabi committed:
Removes the listener integration tests

They cannot cope with the parallelizing of stream additions

amousavigourabi committed:
Address comments

amousavigourabi committed:
Resolve issue with job-stage

amousavigourabi committed:
Replaces for loop by Java fp

amousavigourabi committed:
Removes wrapper

See !105

amousavigourabi committed:
Address reviews

See !105

amousavigourabi committed:
Adds some more documentation

See !105

amousavigourabi committed:
Adds awaits to flaky tests

amousavigourabi committed:
Merge branch '176-streaming' into 'dev'

Resolve "Use Streaming infra instead of lists"

Closes #176

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!105

amousavigourabi committed:
Fix app end guard bug

amousavigourabi committed:
Fixes the naming of clearDirectories

Moves to createDirectories instead to be more reflective of what it does

amousavigourabi committed:
Update Parquet InputFile

As discussed with TA

amousavigourabi committed:
Updates Parquet OutputFile

As discussed with TA

amousavigourabi committed:
Allow DiskOutputFile to overwrite files

amousavigourabi committed:
Update Javadoc

amousavigourabi committed:
Switches to Stream metric aggregation

amousavigourabi committed:
Move KthLargest to core

amousavigourabi committed:
Update CHANGELOG

amousavigourabi committed:
Gets rid of flakes

amousavigourabi committed:
Gets rid of flakes

amousavigourabi committed:
Clean up linearly growing map

amousavigourabi committed:
Fixes thread pool await on app end

amousavigourabi committed:
Add file delete statements

amousavigourabi committed:
Update CHANGELOG

amousavigourabi committed:
Fix tests

amousavigourabi committed:
Merge branch '192-aggregation-metrics-need-to-use-streams' into 'dev'

Resolve "Aggregation metrics need to use Streams"

Closes #192

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!109

amousavigourabi committed:
Resolve overwriting being ignored

amousavigourabi committed:
Updates logs

See !108

amousavigourabi committed:
Adds Spark link

amousavigourabi committed:
Add some documentation on Maven

amousavigourabi committed:
Updates documentation

amousavigourabi committed:
Add sentence on folder clean

amousavigourabi committed:
Adds line on isStageLevel

amousavigourabi committed:
Implement feedback

See !110

amousavigourabi committed:
Update CHANGELOG

amousavigourabi committed:
Fix site skin

amousavigourabi committed:
Merge branch '191-allow-wta-writer-to-overwrite-existing-files' into 'dev'

Resolve "Allow WTA writer to overwrite existing files"

Closes #191

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!108

amousavigourabi committed:
Merge branch '156-docs-umbrealla-issue' into 'dev'

Resolve "Docs Umbrealla issue"

Closes #156

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!110

lyadalachanchu20033 committed:
Merge branch 'dev' into 'main'

Merge to main

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!111

amousavigourabi committed:
Fix typo

amousavigourabi committed:
Merge branch 'fix-maven-site' into 'dev'

Fix typo

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!112

amousavigourabi committed:
Merge branch 'dev' into 'main'

Fix typo that broke Maven

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!113

Commit:
change docs and remove newline

lyadalachanchu20033 committed:
Merge branch '194-remove-new-line-in-proc-cpu-information' into 'dev'

Resolve "Remove New Line in Proc CPU Information"

Closes #194

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!114

amousavigourabi committed:
Moves the map clearing

amousavigourabi committed:
Remove redundant calls

amousavigourabi committed:
Update CHANGELOG

amousavigourabi committed:
Move KeyedStream clean up operation out of loop

amousavigourabi committed:
Change unnecessary visibility modifier

amousavigourabi committed:
Remove obsolete javadoc

amousavigourabi committed:
Add stage level stream guard

amousavigourabi committed:
Remove dummy values to use builder default

amousavigourabi committed:
Fix Stage listener tests

amousavigourabi committed:
Use Awaitility for thread pools in tests

Commit:
give builder of models default values

Commit:
minor shifts

amousavigourabi committed:
More Awaitility

amousavigourabi committed:
Streamlines streams

amousavigourabi committed:
Update POM

amousavigourabi committed:
Streamline kth largest logic

amousavigourabi committed:
Merge branch '195-key-based-map-clears-are-more-inefficient-than-they-need-be' into 'dev'

Resolve "Key based map clears are more inefficient than they need be"

Closes #195

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!115

amousavigourabi committed:
Iterativify the kth largest logic

amousavigourabi committed:
Add CHANGELOG

amousavigourabi committed:
Implement feedback

amousavigourabi committed:
Fix return values

amousavigourabi committed:
Fix linter warnings

amousavigourabi committed:
Merge branch '197-iterative-kth-largest' into 'dev'

Resolve "Iterative kth largest"

Closes #197

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!117

Commit:
resolve merge conflicts

amousavigourabi committed:
Clean up repository structure and add Javadoc

amousavigourabi committed:
Merge remote-tracking branch 'origin/196-final-performance-optimization' into cleanup-repository

amousavigourabi committed:
Remove unnecessary method

amousavigourabi committed:
Removes unnecessary Stream to List conversions

amousavigourabi committed:
Add access modifier and remove branch

amousavigourabi committed:
Simplify POM

amousavigourabi committed:
Fix linter warnings

amousavigourabi committed:
Removes unnecessary excludes from PIT

amousavigourabi committed:
Refactors tests

amousavigourabi committed:
Tweak PIT excludes

Commit:
apply linters

amousavigourabi committed:
Refactors more tests

Commit:
fix test

amousavigourabi committed:
Refactor HDFS tests

amousavigourabi committed:
Clean up the project directory

Commit:
apply suggestions

Commit:
Merge branch '196-final-performance-optimization' into 'dev'

Resolve "Final bug fix and performance optimization"

Closes #196

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!116

amousavigourabi committed:
Refactors tests

amousavigourabi committed:
Fixes critical path

amousavigourabi committed:
Handle Spark failures more nicely

amousavigourabi committed:
Make task level listener more resilient

amousavigourabi committed:
Make listener more resilient

Commit:
apply linters

Commit:
Apply 1 suggestion(s) to 1 file(s)

amousavigourabi committed:
Update CHANGELOG

amousavigourabi committed:
Merge branch 'cleanup-repository' into 'dev'

Cleanup repository

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!118

Commit:
Merge branch 'dev' into 'main'

Merge to main

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!119

Commit:
add spark session close

amousavigourabi committed:
Get rid of some overhead at aggregation

Commit:
update docs

amousavigourabi committed:
Clean up mvn clean

Commit:
add aggregation metric config option

Commit:
remove unnecesary tags from doc strings

Commit:
update javadocs

Commit:
code restyle

Commit:
improve doc strings

Commit:
imporve doc strings

Commit:
beautify doc strings

Commit:
add tests

Commit:
resolve merge conflicts

Commit:
beautify doc strings

Commit:
add config parse test suites

Commit:
update documentation

Commit:
add documentation

Commit:
apply checkstyle fixes

Commit:
apply checkstyle fixes

Commit:
apply linters

Commit:
attempt empty trace debug

amousavigourabi committed:
Use thread pools more safely

Commit:
uncomment EndToEnd lines

Commit:
pass current time to trace filepath as argument

Commit:
tests to increase line coverage

amousavigourabi committed:
apt update

amousavigourabi committed:
Update Maven

amousavigourabi committed:
Fix Maven update

Commit:
add more tests for coverage

Commit:
pull from remote

Commit:
apply linters

Commit:
fix test

Commit:
add tests for coverage

Commit:
apply linters

Commit:
update documentation and change config values

amousavigourabi committed:
Add Stream optimization tests

Commit:
add @since annotation back to doc strings

Commit:
apply linters

Commit:
Merge branch '198-update-docs' into 'dev'

Resolve "Update docs and java doc"

Closes #198

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!120

Commit:
Merge branch 'dev' into 'main'

Merge to main

See merge request cse2000-software-project/2022-2023-q4/cluster-12/extracting-workflow-traces-from-apache-spark-and-enhance-them-with-resource-utilization-information/asml-dev!121

amousavigourabi committed:
Add 'spark-plugin/' from commit 'bfa6f342f9f00ca26021a9251740d92d1515fcf4'

git-subtree-dir: spark-plugin
git-subtree-mainline: e9d505df03fff9bb57208dfb82212977ef5e7ca2
git-subtree-split: bfa6f342f9f00ca26021a9251740d92d1515fcf4

lfdversluis commented:
Could you do a small cleanup of the the .gitlab folder? The bug template, PR template, stages / builds will not work in GitHub. Also files like .gitlab-ci.yaml etc. should be ported or removed.

amousavigourabi committed:
Remove GitLab workflows

amousavigourabi committed:
Rename to be consistent

Kebab case to snake case for the folder to be consistent with the other folders in wta-tools
